### PR TITLE
Add lib declaration for RegExp#unicodeSets

### LINKED ISSUE #57055: Missing RegExp#unicodeSets
### ⚙ Compilation target

ESNext

### ⚙ Library

ESNext

### Missing / Incorrect Definition

Missing `RegExp#unicodeSets` property

### Sample Code

```TypeScript
/[a]/v.unicodeSets
```

### Documentation Link

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/unicodeSets

### PR DESCRIPTION
**Pull request by Josh-Cena**
Fixes #57055

### CODE REVIEW

File: src/lib/esnext.regexp.d.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+interface RegExp {
+    /**
+     * Returns a Boolean value indicating the state of the unicodeSets flag (v) used with with a regular expression.
```
Comment (DanielRosenwasser) [inline src/lib/esnext.regexp.d.ts:3]:
```suggestion
     * Returns a Boolean value indicating the state of the unicodeSets flag (v) used with a regular expression.
```

**Review by sandersn — APPROVED** [review]
This PR is good as-is, but now that it's reviewed, feel free to fix any duped words or typos you want. It's the perfect place to do so.

This should wait until 5.4 beta is over since it's not a bug fix.

**Comment by Josh-Cena** [comment]
I think this is good to go

**Comment by sandersn** [comment]
Some tests failed last time CI ran. Does this need to have main merged into it? Or is it the code itself that is failing?

**Comment by Josh-Cena** [comment]
Mmm, apparently there are other config files that need to be changed but I'm unaware of.

**Comment by Josh-Cena** [comment]
Wow, the number of baselines updated by an extra lib is surprising.

**Comment by Josh-Cena** [comment]
@sandersn Now that 5.4 is out, will this be merged soon?

**Pull request merged**